source/extensions/filters/common/ext_authz/check_request_utils.cc

```diff
@@ -134,25 +142,33 @@ void CheckRequestUtils::setHttpRequest(
 
   // Fill in the headers.
   auto* mutable_headers = httpreq.mutable_headers();
+  auto* mutable_header_map = httpreq.mutable_header_map();
```

Comment:
This creates a confusing protobuf message: both fields exist in the message `msg.has_header_map() && msg.has_headers()`. Please refactor so that the message comes out with only the desired message set, and add tests for this.